firedrake/preconditioners/pmg.py

```diff
@@ -303,18 +453,20 @@ def tensor_product_space_query(V):
     if isinstance(ele, firedrake.TensorProductElement):
         family = set(e.family() for e in ele.sub_elements())
         try:
-            variant, = set(e.variant() or "spectral" for e in ele.sub_elements())
```

Comment:
Can you comment on what the changes around here are doing?

Reply:
For one moment I thought there was a bug here, after checking twice, I think that it should be kept as it was before.

Reply:
Can  you just add a comment explaining what's going on? (Especially if you got confused yourself!)